src/main/scala/chisel3/internal/firrtl/Emitter.scala

```diff
@@ -83,17 +80,10 @@ private class Emitter(circuit: Circuit) {
     */
   private def emit(m: Component): String = {
     // Generate the body.
-    val defn = moduleDefn(m)
-
-    defnMap get (m.id.desiredName, defn) match {
-      case Some(duplicate) =>
-        m.id setModName duplicate.name
-        ""
-      case None =>
-        defnMap((m.id.desiredName, defn)) = m
-        m.id setModName m.name
-        moduleDecl(m) + defn
-    }
+    val sb = new StringBuilder
+    sb append moduleDecl(m)
+    sb append moduleDefn(m)
```

Comment:
These should use dot notation
Infix notation should only be used for purely functional methods (http://docs.scala-lang.org/style/method-invocation.html#arity-1)